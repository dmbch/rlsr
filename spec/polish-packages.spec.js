/* eslint-env node, jest */

const polishPackages = require('../src/transform/polish-packages');
const getEnv = require('./fixtures/env-small.fixture');

describe('polishPackages()', () => {
  it('enriches packages with base data', () => {
    expect(
      polishPackages(getEnv({ a: { name: 'a' }, b: { name: 'b' } }, []))
    ).toMatchSnapshot();
  });
});
