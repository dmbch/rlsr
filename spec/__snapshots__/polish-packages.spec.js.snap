// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`polishPackages() enriches packages with base data 1`] = `
Object {
  "appRoot": "/",
  "changelog": Object {},
  "consts": Object {
    "levels": Array [
      "patch",
      "minor",
      "major",
    ],
    "nsp": "rlsr",
    "rlsrLatest": "rlsr-latest",
  },
  "dbg": [Function],
  "log": [Function],
  "mainPackage": Object {
    "rlsr": Object {},
    "version": "1.2.3",
  },
  "messages": Array [],
  "packages": Object {
    "a": Object {
      "name": "a",
      "rlsr": Object {
        "dependencies": Array [],
        "determinedIncrementLevel": -1,
        "devDependencies": Array [],
        "devRelations": Array [],
        "messages": Array [],
        "peerDependencies": Array [],
        "peerRelations": Array [],
        "relatedMessages": Array [],
        "relations": Array [],
      },
    },
    "b": Object {
      "name": "b",
      "rlsr": Object {
        "dependencies": Array [],
        "determinedIncrementLevel": -1,
        "devDependencies": Array [],
        "devRelations": Array [],
        "messages": Array [],
        "peerDependencies": Array [],
        "peerRelations": Array [],
        "relatedMessages": Array [],
        "relations": Array [],
      },
    },
  },
}
`;
